src/apps/integrations/loris/service/integrations.py

```diff
@@ -0,0 +1,43 @@
+from apps.integrations.crud.integrations import IntegrationsCRUD
+from apps.integrations.db.schemas.schemas import IntegrationsSchema
+from apps.integrations.domain import AvailableIntegrations
+from apps.integrations.errors import UnsupportedIntegrationError
+from apps.integrations.loris.domain.loris_integrations import IntegrationsCreate
+from apps.integrations.loris.domain.loris_projects import LorisProjects
+from apps.integrations.loris.service.loris_client import LorisClient
+
+__all__ = [
+    "IntegrationsService",
+]
+
+
+class IntegrationsService:
+    def __init__(self, session) -> None:
+        self.session = session
+
+    async def create_integration(
+        self,
+        type: str,
+        params: IntegrationsCreate,
+    ) -> LorisProjects:
+        if type not in AvailableIntegrations.as_list():
+            raise UnsupportedIntegrationError(type=type)
+
+        token = await LorisClient.login_to_loris(params.hostname, params.username, params.password)
+        projects_raw = await LorisClient.list_projects(params.hostname, token)
+
+        await IntegrationsCRUD(self.session).create(
+            IntegrationsSchema(
+                applet_id=params.applet_id,
+                type=type,
+                configuration={
+                    "hostname": params.hostname,
+                    "username": params.username,
+                    "password": params.password,
+                },
+            )
+        )
+
+        projects = list(projects_raw["Projects"].keys())
+
+        return LorisProjects(projects=projects)
```

Comment:
I think this would work better inside a match case statement, with the default case raising the exception.

```
match type:
  case AvailableIntegrations.LORIS:
  ...
```

Reply:
Thank you for the comments. Definitely looking inside a map is not efficient at all.
The current changes followed your suggestions hoping that this pattern matching does not need to be repeated somewhere else.
But I will check if it is possible to threat this as a set instead given that it has better lookup performance.